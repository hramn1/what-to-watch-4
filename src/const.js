export const ALL_GENRES = `All genres`;
export const SHOW_FILMS = 8;
export const Time = {
  MINUTES_IN_HOUR: 60,
  SECONDS_IN_MINUTE: 60,
};
export const Pages = {
  ADD_REVIEW: `/review`,
  FILM: `/films`,
  MAIN: `/`,
  MY_LIST: `/mylist`,
  LOGIN: `/login`,
  PLAYER: `/player`
};
