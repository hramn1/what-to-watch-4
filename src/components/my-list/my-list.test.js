import React from "react";
import renderer from "react-test-renderer";
import {Provider} from 'react-redux';
import thunk from 'redux-thunk';
import configureStore from 'redux-mock-store';
import {ALL_GENRES} from '../../const.js';
import MyList from "./my-list.jsx";
import NameSpace from '../../reducer/name-space.js';
import {Router} from 'react-router-dom';
import history from '../../history.js';
import {createAPI} from '../../api.js';

const api = createAPI();

const mockStore = configureStore([
  thunk.withExtraArgument(api)
]);
const availableGenres = [`crime`, `drama`];
const favoriteFilms = [
  {
    id: `1`,
    title: `Fantastic Beasts: The Crimes of Grindelwald`,
    poster: `img/fantastic-beasts-the-crimes-of-grindelwald.jpg`,
  },
  {
    id: `2`,
    title: `Bohemian Rhapsody`,
    poster: `img/bohemian-rhapsody.jpg`,
  },
];
const cardFilms = {
  title: `The Grand Budapest Hotel`,
  genre: `Drame`,
  year: `2014`,
};
const showFilms = 2;
const authorizationStatus = `NO-AUTH`;
const authorizationInfo = {};
describe(`MyList`, () => {
  const store = mockStore({
    [NameSpace.DATA]: {
      cardFilms,
      availableGenres,
      favoriteFilms,
    },
    [NameSpace.APP]: {
      currentGenre: ALL_GENRES,
      showFilms,
    },
    [NameSpace.USER]: {
      authorizationStatus,
      authorizationInfo
    },
  });

  it(`Render Mylist`, () => {
    const tree = renderer.create(
        <Router history={history}>
          <Provider store={store}>
            <MyList
              onFilmCardClick={() => {}}
              onPlayClick = {()=>{}}
              handleFilmFavorite= {()=>{}}
              loadFavoriteMovies={()=>{}}
              films={favoriteFilms}
            />
          </Provider>
        </Router>, {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
