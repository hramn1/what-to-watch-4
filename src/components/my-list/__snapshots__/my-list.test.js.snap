// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyList Render Mylist 1`] = `
<div
  className="user-page"
>
  <header
    className="page-header user-page__head"
  >
    <div
      className="logo"
    >
      <a
        className="logo__link"
        href="/"
        onClick={[Function]}
      >
        <span
          className="logo__letter logo__letter--1"
        >
          W
        </span>
        <span
          className="logo__letter logo__letter--2"
        >
          T
        </span>
        <span
          className="logo__letter logo__letter--3"
        >
          W
        </span>
      </a>
    </div>
    <h1
      className="page-title user-page__title"
    >
      My list
    </h1>
    <div
      className="user-block"
    >
      <div
        className="user-block__avatar"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <img
            height="63"
            width="63"
          />
        </a>
      </div>
    </div>
  </header>
  <section
    className="catalog"
  >
    <h2
      className="catalog__title visually-hidden"
    >
      Catalog
    </h2>
    <div
      className="catalog__movies-list"
    >
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            width="280"
          >
            <source
              type="video/webm; codecs=\\"vp8, vorbis\\""
            />
            <source
              type="video/ogg; codecs=\\"theora, vorbis\\""
            />
            <source
              type="video/mp4; codecs=\\"avc1.42E01E, mp4a.40.2\\""
            />
          </video>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/1"
            onClick={[Function]}
          >
            Fantastic Beasts: The Crimes of Grindelwald
          </a>
        </h3>
      </article>
      <article
        className="small-movie-card catalog__movies-card"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseOver={[Function]}
      >
        <div
          className="small-movie-card__image"
        >
          <video
            height="175"
            width="280"
          >
            <source
              type="video/webm; codecs=\\"vp8, vorbis\\""
            />
            <source
              type="video/ogg; codecs=\\"theora, vorbis\\""
            />
            <source
              type="video/mp4; codecs=\\"avc1.42E01E, mp4a.40.2\\""
            />
          </video>
        </div>
        <h3
          className="small-movie-card__title"
        >
          <a
            className="small-movie-card__link"
            href="/films/2"
            onClick={[Function]}
          >
            Bohemian Rhapsody
          </a>
        </h3>
      </article>
    </div>
    <footer
      className="page-footer"
    >
      <div
        className="logo"
      >
        <a
          className="logo__link logo__link--light"
          href="/"
          onClick={[Function]}
        >
          <span
            className="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            className="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            className="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        className="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </section>
</div>
`;
